### PR TITLE
Definitions for brotli-js

### PR DESCRIPTION
**Pull request by hu0529**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by hu0529** [comment]
Definitions for brotli-js

File: types/brotli-js/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for brotli-js 1.0.2
+// Project: https://github.com/ganruip/brotli-js
+
+export function compressArray(bufView: Uint8Array | ArrayBuffer, compressinLevel: number ): any;
```
Comment (codershiba) [inline types/brotli-js/index.d.ts:4]:
```suggestion
export function compressArray(buffer: Uint8Array | ArrayBuffer, level: number): any;
```

File: types/brotli-js/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+export function decompressArray(compressData: Uint8Array | ArrayBuffer): any;
```
Comment (codershiba) [inline types/brotli-js/index.d.ts:6]:
```suggestion
export function decompressArray(buffer: Uint8Array | ArrayBuffer): any;
```

**Comment by hu0529** [comment]
Definitions for brotli-js

**Comment by hu0529** [comment]
Definitions for brotli-js

**Comment by hu0529** [comment]
Definitions for brotli-js

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/brotli-js/brotli-js-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against cff859a964806704aaeb5fc64ba0cd944180c16b
</p>

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Comment by hu0529** [comment]
Please review. Thank you.

**Comment by hu0529** [comment]
Please review. Thank you.

**Comment by hu0529** [comment]
> Re-ping @«anyone?»:
> 
> This PR has been out for over a week, yet I haven't seen any reviews.
> 
> Could someone please give it some attention? Thanks!

Please review. Thank you.

**Comment by RyanCavanaugh** [comment]
Rejecting this based on obscurity. Brotli compression is built into node.js via the `zlib` library, and there are two other packages (`brotli` and `compress-brotli`) which have 2.4M and 400K weekly downloads, respectively, compared to `brotli-ji` which has a dozen downloads on a good week. I don't see any reason someone would be using this old library compared to the alternatives, and indeed very few people are.

**Pull request closed without merging**